ocaml/tests/test_tracing.ml

```diff
@@ -135,11 +135,90 @@ let test_tracing_exn_backtraces () =
+    )
+  in
+  Alcotest.(check bool)
+    "Good key, value pair" true
+    (Tracing.validate_attribute good) ;
```

Comment:
I recommend instead to have a test specification: list of (description, value), then have List.iter to test all of them:

```ocaml
let good_attributes = [("number", ("foo1", "bar")); ...] in
let bad_attributes = [(...); ...] in

let test_good_attribute (desc, attr) =
  Alcotest.(check bool) ("Good key, value pair with " ^ desc) true attr
in

let test_bad_attribute (desc, attr) =
  Alcotest.(check bool) ("Bad key, value pair with " ^ desc) false attr
in

List.iter test_good_attribute good_attributes ;
List.iter test_bad_attribute bad_attributes 
```

Reply:
https://github.com/xapi-project/xen-api/pull/4966/commits/c3d65098735490561bb147364b4043ff79f045d8